torch/_inductor/pattern_matcher.py

```diff
@@ -251,14 +251,73 @@ def replace_by_example(
             else contextlib.nullcontext()
         )
 
+        def should_propagate_arg_kwarg_vals(nodes: list[torch.fx.Node]) -> bool:
+            if len(nodes) != 1:
+                return False
+            node = nodes[0]
+            if "arg_kwarg_vals" not in node.meta:
+                return False
+            return node.target in OrderedSet(
+                [
+                    torch.ops.higher_order.triton_kernel_wrapper_functional,
+                ]
+            )
+
         with context:
             if trace_fn is None:
                 trace_fn = functools.partial(
                     fwd_only, run_functional_passes=run_functional_passes
                 )
-            replacement = trace_fn(
-                replacement_fn, torch.fx.map_arg(args, lambda arg: arg.meta["val"])
-            )
+
+            if should_propagate_arg_kwarg_vals(self.nodes):
+                # Our strategy is:
+                # 1) trace out the graph with arg_kwarg_vals (which have accurate eager-mode metadata)
+                # 2) trace out the graph with vals (which have the accurate Inductor metadata)
```

Comment:
We should be able to short-circuit this is they are equivalent metadata.

Reply:
What do you mean?